caikit/runtime/train.py

```diff
@@ -77,10 +80,38 @@ def write_termination_log(text: str, log_file="/dev/termination-log"):
         )
 
 
+# Final tasks before exiting the container
+def exit_complete(
+    exit_code: int,
+    save_path: str,
+    message: str,
+    termination_log_file: str,
+    enable_termination_log: bool,
+):
+    if exit_code != 0:
+        write_termination_log(message, termination_log_file, enable_termination_log)
+
+    if save_path:
+        try:
+            complete_path = os.path.join(save_path, ".complete")
+            log.info(f"Creating completion file at: {complete_path}")
+            Path(complete_path).touch()
+        except Exception as e:
+            log.warning(f"Unable to write completion file due to execption: {e}")
```

Comment:
Lazy log interpolation

```suggestion
            log.info("Creating completion file at: %s", complete_path)
            Path(complete_path).touch()
        except Exception as e:
            log.warning("Unable to write completion file due to execption: %s", e)
```